File: Python/addtime/addtime.py
# Python 3.5.2
import sys

class Time:
        def __init__(self, hrs, mins):
                self.hrs = hrs
                self.mins = mins
                
        def __repr__(self):
                return "{0:02d}:{1:02d}".format(self.hrs, self.mins)
                
        def __add__(self, t):
                offset = self.hrs * 60 + self.mins
                tot = t + offset
                return Time(tot // 60, tot % 60)

if len(sys.argv) == 1:
	Istr = input("Start time:	")
	Istr = Istr.replace(' ', ':').split(':', 2)
else:
	Istr = sys.argv[1].split(':')
	if len(sys.argv) >= 3:
		Istr[len(Istr):] = [sys.argv[2]]

Istr = [int(x) for x in Istr]

if len(Istr) == 2:  # Supplement input
	elapse = eval(input("Period(min):	"))
else:  # Use existing
	elapse = Istr[2]
	
start = Time(Istr[0], Istr[1])

Ostr_S = "Start:	{0}".format(start)
Ostr_E = "End:	{0}".format(start + elapse)

print("{0}	{1:+}min\n{2}".format(Ostr_S, elapse, Ostr_E))

File: Cpp/addtime/addtime.cpp
#include <string>
#include <iostream> // showpos/noshowpos
#include <iomanip> // setw, setfill
#include <math.h> // floor

// compile with "g++ -std=c++17"
using namespace std;

class Time {
	public:
		int hrs;
		int mins;
		Time(int hrs_, int mins_) {
			hrs = hrs_;
			mins = mins_;
		}
};

Time operator+(const Time lhs, const int rhs) {
	int offset = lhs.hrs * 60 + lhs.mins;
	int tot = rhs + offset;
	return Time(floor(tot / 60), tot % 60);
}		

ostream& operator<<(ostream& stream, Time a) {
	stream << noshowpos << setfill('0') << setw(2) << a.hrs << ':' 
		<< setw(2) << a.mins;
	return stream;
}

int main(int argc, char** argv) {
	if(argc < 3) {
		cout << "Error:\thh:mm mins expected." << endl;
		return 1;
	}
	string strT = argv[1];
	int delimpos = strT.find(':');
	string str_hrs = strT.substr(0, delimpos);
	string str_mins = strT.substr(delimpos + 1, strT.length());
	Time start(stoi(str_hrs), stoi(str_mins));
	int elapse = atoi(argv[2]);
	
	cout << "Start:\t" << start << "\t"
		<<  showpos << elapse << "\n"
		<< "End:\t" << start + elapse << endl;
	return 0;
}
